cpp/core/jni/JniWrapper.cc

```diff
@@ -1188,13 +1193,14 @@ JNIEXPORT jlong JNICALL Java_io_glutenproject_vectorized_ColumnarBatchSerializer
     JNIEnv* env,
     jobject,
     jlong cSchema,
-    jlong allocId) {
+    jlong memoryManagerId) {
   JNI_METHOD_START
-  auto* allocator = reinterpret_cast<std::shared_ptr<MemoryAllocator>*>(allocId);
-  GLUTEN_DCHECK(allocator != nullptr, "Memory pool does not exist or has been closed");
+  MemoryManager* memoryManager = reinterpret_cast<MemoryManager*>(memoryManagerId);
+  GLUTEN_DCHECK(memoryManager != nullptr, "Memory manager does not exist or has been closed");
+  auto arrowPool = asArrowMemoryPool(memoryManager->getMemoryAllocator());
   auto backend = createBackend();
   auto serializer =
-      backend->getColumnarBatchSerializer((*allocator).get(), reinterpret_cast<struct ArrowSchema*>(cSchema));
+      backend->getColumnarBatchSerializer(memoryManager, arrowPool, reinterpret_cast<struct ArrowSchema*>(cSchema));
```

Comment:
Would you like to open another ticket for including arrow memory pool to memory manager? Let's fix that later.

Reply:
https://github.com/oap-project/gluten/issues/2853